core/domain/user_jobs_one_off_test.py

```diff
@@ -34,6 +34,107 @@
+        # (This is used to check that there are no duplicate 
+        # entries in the contribution lists.)
+        self.signup(self.USER_A_EMAIL, self.USER_A_USERNAME)
+        self.user_a_id = self.get_user_id_from_email(self.USER_A_EMAIL)
+        self.signup(self.USER_B_EMAIL, self.USER_B_USERNAME)
+        self.user_b_id = self.get_user_id_from_email(self.USER_B_EMAIL)
+        self.signup(self.USER_C_EMAIL, self.USER_C_USERNAME)
+        self.user_c_id = self.get_user_id_from_email(self.USER_C_EMAIL)
+        self.signup(self.USER_D_EMAIL, self.USER_D_USERNAME)
+        self.user_d_id = self.get_user_id_from_email(self.USER_D_EMAIL)
+
+        self.save_new_valid_exploration(
+            self.EXP_ID_1, self.user_b_id, end_state_name='End')
+
+        exp_services.update_exploration(self.user_c_id, self.EXP_ID_1, [{
+            'cmd': 'edit_exploration_property',
+            'property_name': 'objective',
+            'new_value': 'the objective'
+        }], 'Test edit')
+
+        self.save_new_valid_exploration(
+            self.EXP_ID_2, self.user_d_id, end_state_name='End')
+
+        exp_services.update_exploration(self.user_d_id, self.EXP_ID_2, [{
+            'cmd': 'edit_exploration_property',
+            'property_name': 'objective',
+            'new_value': 'the objective'
+        }], 'Test edit')
+
+    def test_null_case(self):
+        """Tests the case where user has no created or edited explorations."""
+
+        self._run_one_off_job()
+        user_a_contributions_model = user_models.UserContributionsModel.get(
+            self.user_a_id, strict=False)
+        self.assertEqual(user_a_contributions_model.created_exploration_ids, [])
+        self.assertEqual(user_a_contributions_model.edited_exploration_ids, [])
+
+    def test_created_exp(self):
+        """Tests the case where user has created (and therefore edited) 
+        an exploration."""
+
+        self._run_one_off_job()
+
+        user_b_contributions_model = user_models.UserContributionsModel.get(
+            self.user_b_id)
+
+        self.assertEqual(
+            user_b_contributions_model.created_exploration_ids, [self.EXP_ID_1])
+        self.assertEqual(
+            user_b_contributions_model.edited_exploration_ids, [self.EXP_ID_1])
+
+    def test_edited_exp(self):
+        """Tests the case where user has an edited exploration."""
+
+        self._run_one_off_job()
+        user_c_contributions_model = user_models.UserContributionsModel.get(
+            self.user_c_id)
+        self.assertEqual(
+            user_c_contributions_model.edited_exploration_ids, [self.EXP_ID_1])
+
+    def test_for_duplicates(self):
+        """Tests the case where user has an edited exploration, and edits
+        it again making sure it is not duplicated."""
+
+        self._run_one_off_job()
+        user_d_contributions_model = user_models.UserContributionsModel.get(
+            self.user_d_id)
+        self.assertEqual(
```

Comment:
Just realized: should check equality of the output here, not lengths, same as you do in the other tests.